### PR TITLE
cleanup(webpack): remove previously deprecated options

### PR DESCRIPTION
**Pull request by jaysoo**
This PR removes `@nrwl/webpack:webpack` options that were previously deprecated and haven't supported in a long time.

* `maxWorkers` - This was passed to `fork-ts-checker-webpack-plugin` previously, but it was removed over a year ago since that plugin removed support for it.
* `budgets` - This support was removed a few major versions ago in Nx but we never removed it from the schema. It does nothing.

Does not have any impact on users.

### CODE REVIEW

**Review by barbados-clemens — COMMENTED** [review]
do any of these options need to have a migration or should be deprecated first before removal?

File: docs/generated/manifests/menus.json [hunk 1]
```diff
@@ -2684,6 +2684,14 @@
...
             "children": [],
             "disableCollapsible": false
+          },
+          {
+            "name": "Identify Dependencies Between Folders",
+            "path": "/recipes/other/identify-dependencies-between-folders",
+            "id": "identify-dependencies-between-folders",
+            "isExternal": false,
+            "children": [],
+            "disableCollapsible": false
```
Comment (barbados-clemens) [inline docs/generated/manifests/menus.json:2694]: These seem unrelated? maybe needs to be rebases with master?

Comment (jaysoo) [reply docs/generated/manifests/menus.json:2694]: ah yeah, i can try that

**Review by barbados-clemens — APPROVED** [review]
